pkg/occlient/templates.go

```diff
@@ -440,11 +440,25 @@ func addBootstrapVolume(dc *appsv1.DeploymentConfig, dcName string) {
 // dc is the deployment config to be updated
 // dcName is the name of the deployment config
 func addBootstrapVolumeMount(dc *appsv1.DeploymentConfig, dcName string) {
+	addVolumeMount(dc, getAppRootVolumeName(dcName), "/opt/app-root", "app-root")
+}
+
+// addDepoymentDirVolumeMount mounts the bootstrap volume to the deployment config
+// in a sub path where the ODO_S2I_DEPLOYMENT_DIR is present for optimisation purposes
+// dc is the deployment config to be updated
+// dcName is the name of the deployment config
+func addDepoymentDirVolumeMount(dc *appsv1.DeploymentConfig, mountPath string) {
+	addVolumeMount(dc, getAppRootVolumeName(dc.Name), mountPath, "deployment")
```

Comment:
why are we passing in "deployment" ? should be from a constant / more unique name?

Reply:
I followed the same pattern as the `addBootstrapVolumeMount` method

Reply:
I also think "deployment" can be a constant.

Reply:
done